core/controllers/classifier.py

```diff
@@ -123,3 +129,61 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+def update_failed_jobs(job_models):
+    for job_model in job_models:
+        classifier_services.mark_training_job_failed(job_model.id)
+
+
+def fetch_training_data(exp_id, state):
+    exp_model = exp_services.get_exploration_by_id(exp_id)
+    state_model = exp_model.states[state]
+    training_data = state_model.get_training_data()
+    return training_data
+
+
+def fetch_next_job():
+    classifier_job_models = (
+        classifier_services.get_all_classifier_training_jobs())
```

Comment:
I think this whole function can be implemented in a better way. First of all use GAE's NDB [Query](https://cloud.google.com/appengine/docs/standard/python/ndb/queries) to fetch models, sorted according to `created_on`, so that you don't have to sort afterwards. Also only fetch those jobs whose status is PENDING or NEW.

So fetch some 10 models that meets above criteria. Go through jobs one by one until you find a job with NEW status. If you encounter any PENDING job for which `(datetime.datetime.utcnow() - (classifier_job_model.last_updated) < TTL)` is True then mark them FAILED. If you don't find a NEW job in first 10 models then fetch next 10. For this you can use [Cursor](https://cloud.google.com/appengine/docs/standard/python/ndb/queries#cursors) for bookkeeping. 

That way we take care of FAILED job and we also get our new job without retrieving all models from DB.